modules/phase_field/src/base/PhaseFieldApp.C

```diff
@@ -161,6 +162,8 @@ PhaseFieldApp::registerApps()
   registerApp(PhaseFieldApp);
 }
 
+// External entry point for dynamic object registration
+extern "C" void PhaseFieldApp__registerObjects(Factory & factory) { PhaseFieldApp::registerObjects(factory); }
```

Comment:
I was thinking even more granular than that.  Why not just call out the name of the object you want?